app/rpc/rpccontext/notificationmanager.go

```diff
@@ -338,11 +348,33 @@ func (nl *NotificationListener) convertUTXOChangesToUTXOsChangedNotification(
 				notification.Removed = append(notification.Removed, utxosByAddressesEntries...)
 			}
 		}
+	} else {
+		for scriptPublicKeyString, addedPairs := range utxoChanges.Added {
+			addressString := nl.scriptPubKeyStringToAddressString(scriptPublicKeyString)
+			utxosByAddressesEntries := ConvertUTXOOutpointEntryPairsToUTXOsByAddressesEntries(addressString, addedPairs)
+			notification.Added = append(notification.Added, utxosByAddressesEntries...)
+		}
+		for scriptPublicKeyString, removedOutpoints := range utxoChanges.Removed {
+			addressString := nl.scriptPubKeyStringToAddressString(scriptPublicKeyString)
+			utxosByAddressesEntries := convertUTXOOutpointsToUTXOsByAddressesEntries(addressString, removedOutpoints)
+			notification.Removed = append(notification.Removed, utxosByAddressesEntries...)
+		}
 	}
 
 	return notification
 }
 
+func (nl *NotificationListener) scriptPubKeyStringToAddressString(scriptPublicKeyString utxoindex.ScriptPublicKeyString) string {
+	scriptPubKey := utxoindex.ConvertStringToScriptPublicKey(scriptPublicKeyString)
+	_, address, err := txscript.ExtractScriptPubKeyAddress(scriptPubKey, nl.params)
+	var addressString string
+	if err != nil { // This just means the script is not standard
```

Comment:
`ExtractScriptPubKeyAddress` should just return `NonStandardTy` for non standard scripts, and no error